prow/config/prow-config-documented.yaml

```diff
@@ -0,0 +1,575 @@
+                        strict: false
+                    restrictions:
+                        teams:
+                          - ""
+                        users:
+                          - ""
+            required_linear_history: false
+            required_pull_request_reviews:
+                dismiss_stale_reviews: false
+                dismissal_restrictions:
+                    teams:
+                      - ""
+                    users:
+                      - ""
+                require_code_owner_reviews: false
+                required_approving_review_count: 0
+            required_status_checks:
+                contexts:
+                  - ""
+                strict: false
+            restrictions:
+                teams:
+                  - ""
+                users:
+                  - ""
+    protect: false
+    required_linear_history: false
+    required_pull_request_reviews:
+        dismiss_stale_reviews: false
+        dismissal_restrictions:
+            teams:
+              - ""
+            users:
+              - ""
+        require_code_owner_reviews: false
+        required_approving_review_count: 0
+    required_status_checks:
+        contexts:
+          - ""
+        strict: false
+    restrictions:
+        teams:
+          - ""
+        users:
+          - ""
+deck:
+    # AdditionalAllowedBuckets is a list of storage buckets to allow in artifact requests
+    # (in addition to those listed in the GCSConfiguration).
+    # Setting this field requires "SkipStoragePathValidation" also be set to `false`.
+    additional_allowed_buckets:
+      - ""
+
+    # Branding of the frontend
+    branding:
+        # BackgroundColor is the color of the background.
+        background_color: ' '
+
+        # Favicon is the location of the favicon that will be loaded in deck.
+        favicon: ' '
+
+        # HeaderColor is the color of the header.
+        header_color: ' '
+
+        # Logo is the location of the logo that will be loaded in deck.
+        logo: ' '
+
+    # ExternalAgentLogs ensures external agents can expose
+    # their logs in prow.
+    external_agent_logs:
+      - # Agent is an external prow agent that supports exposing
+        # logs via deck.
+        agent: ' '
+
+        # SelectorString compiles into Selector at load time.
+        selector: ' '
+
+        # URLTemplateString compiles into URLTemplate at load time.
+        url_template: ' '
+
+    # GoogleAnalytics, if specified, include a Google Analytics tracking code on each page.
+    google_analytics: ' '
+
+    # HiddenRepos is a list of orgs and/or repos that should not be displayed by Deck.
+    hidden_repos:
+      - ""
+
+    # Deprecated: RerunAuthConfig specifies who is able to trigger job reruns if that feature is enabled.
+    # The permissions here apply to all jobs.
+    # This option will be removed in favor of RerunAuthConfigs in July 2020.
+    rerun_auth_config:
+        github_orgs:
```

Comment:
Are these not commented in the source?

Reply:
Yeah, see https://github.com/kubernetes/test-infra/pull/19702#discussion_r512399362

The basic issue is that genyaml considers each file to be its own package which means that if you have a struct in file a that references another struct in file b it won't work. Ref: https://github.com/kubernetes/test-infra/blob/30c5d23c40dbf10ce2b4ed9232a6199c385923b1/pkg/genyaml/genyaml.go#L154

Thats a genyaml bug though that wasn't introduced by this PR, so I'd like to fix it in a follow-up.